lib/spack/spack/cmd/ci.py

```diff
@@ -515,42 +515,98 @@ def ci_rebuild(args):
     # No hash match anywhere means we need to rebuild spec
 
     # Start with spack arguments
-    install_args = [base_arg for base_arg in CI_REBUILD_INSTALL_BASE_ARGS]
+    spack_cmd = [SPACK_COMMAND, "--color", "always", "--verbose"]
 
     config = cfg.get("config")
     if not config["verify_ssl"]:
-        install_args.append("-k")
+        spack_cmd.append("-k")
 
-    install_args.extend(
-        [
-            "install",
-            "--show-log-on-error",  # Print full log on fails
-            "--keep-stage",
-            "--use-buildcache",
-            "dependencies:only,package:never",
-        ]
-    )
+    install_args = [
+        "--show-log-on-error",  # Print full log on fails
+    ]
 
     can_verify = spack_ci.can_verify_binaries()
     verify_binaries = can_verify and spack_is_pr_pipeline is False
     if not verify_binaries:
         install_args.append("--no-check-signature")
 
+    cdash_args = []
     if cdash_handler:
         # Add additional arguments to `spack install` for CDash reporting.
-        install_args.extend(cdash_handler.args())
+        cdash_args.extend(cdash_handler.args())
 
     # A compiler action of 'FIND_ANY' means we are building a bootstrap
     # compiler or one of its deps.
     # TODO: when compilers are dependencies, we should include --no-add
     if compiler_action != "FIND_ANY":
         install_args.append("--no-add")
 
-    # Identify spec to install by hash
-    install_args.append("/{0}".format(job_spec.dag_hash()))
+    # TODO: once we have the concrete spec registry, use the DAG hash
+    # to identify the spec to install, rather than the concrete spec
+    # json file.
+
+    slash_hash = "/{}".format(job_spec.dag_hash())
+    deps_install_args = install_args
+    root_install_args = install_args + [
+        "--keep-stage",
+        "--only=package",
+        "--use-buildcache=package:never,dependencies:only",
+        slash_hash,
+    ]
+
+    commands = [
+        # apparently there's a race when spack bootstraps? do it up front once
```

Comment:
Why did I think the runner images had everything and didn't need any bootstrapping? I suppose spack moves so fast that's not a practical goal?

Reply:
It's part of https://github.com/spack/spack/pull/32262, some distro's provide patchelf versions that are a decade old, and manipulating binaries is very bug-prone, so it makes sense to stick to a recent patchelf.